extension/chrome/elements/compose-modules/compose-recipients-module.ts

```diff
@@ -555,19 +561,19 @@ export class ComposeRecipientsModule extends ViewModule<ComposeView> {
       const contacts: ContactPreview[] = await ContactStore.search(undefined, { substring });
       this.view.errModule.debug(`searchContacts substring: ${substring}`);
       this.view.errModule.debug(`searchContacts db count: ${contacts.length}`);
-      this.renderSearchRes(input, contacts, { substring });
+      await this.renderSearchRes(input, contacts, { substring });
       if (contacts.length >= this.MAX_CONTACTS_LENGTH) {
         this.view.errModule.debug(`searchContacts 2, count: ${contacts.length}`);
         return;
       }
       let foundOnGoogle: EmailProviderContact[] = [];
-      if (this.googleContactsSearchEnabled) {
+      if (await this.googleContactsSearchEnabled) {
```

Comment:
I also wonder how the UI would look like if this request hangs for a minute, for example. Should we introduce some sort of timeout? Or run this block concurrently with `searchContacts 4`?

Reply:
Timeout makes sense to me
Maybe need to update `api` infra to accept optional timeout param and pass it to $.ajax if passed?
https://github.com/FlowCrypt/flowcrypt-browser/blob/01c1e0087013661c1df82a0435f070d3f0cb7354/extension/js/common/api/shared/api.ts#L73

Reply:
If the request times out, should we try to search contacts? E.g. introduce a third state (disabled, enabled, maybe-enabled)?

Reply:
I think so

Reply:
We agreed to search google contacts on `undefined` too.